test/WebApiShimFw46.FunctionalTests/FunctionalTest/TelemetryModuleWorkingWebApiTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace SampleWebAppIntegration.FunctionalTest
+{
+    using System.Linq;
+    using System.Net.Http;
+    using System.Threading;
+    using FunctionalTestUtils;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Xunit;
+
+    public class TelemetryModuleWorkingWebApiTests : TelemetryTestsBase
+    {
+        private const string assemblyName = "WebApiShimFw46.FunctionalTests";
+
+#if dnx451
+
+        [Fact]
+        public void TestBasicDependencyPropertiesAfterRequestingBasicPage()
+        {
+            using (var server = new InProcessServer(assemblyName))
+            {
+                const string RequestPath = "/api/values";
+
+                var expectedDependencyTelemetry = new DependencyTelemetry();
+                expectedDependencyTelemetry.Name = server.BaseHost + RequestPath;
+                expectedDependencyTelemetry.ResultCode = "200";
+                expectedDependencyTelemetry.Success = true;
+                this.ValidateBasicDependency(server, RequestPath, expectedDependencyTelemetry);
+            }
+        }
+
+        [Fact]
+        public void TestIfPerformanceCountersAreCollected()
+        {
+            using (var server = new InProcessServer(assemblyName))
+            {
+                const string RequestPath = "/";
+                var httpClient = new HttpClient();
+                var task = httpClient.GetAsync(server.BaseHost + RequestPath);
+                task.Wait(TestTimeoutMs);
+                var result = task.Result;
+                Thread.Sleep(70000);
```

Comment:
Just to make sure - this is not blocking this release. However let's discuss if we can mock the environment in the test so we don't have to wait this long - this is simply not practical for development especially as unit test base starts to grow.